### ISSUE TITLE
[no-unresolved] False positive when importing type from DefinitelyTyped

### ISSUE THREAD

**Issue by ryb73** [import-js/eslint-plugin-import#2135]
I get the following false positive:

```
  5:27  error  Unable to resolve path to module 'chromecast-caf-receiver'  import/no-unresolved
```

on this line:

```ts
import type { cast } from "chromecast-caf-receiver";
```

I have `@types/chromecast-caf-receiver` included in package.json and installed.

I'm using:
eslint-plugin-import 2.23.4
eslint 7.29.0
Yarn Berry with workspaces, node_modules linker

**Comment by ljharb** [comment]
I believe this should be solved in v2.24.0. Happy to reopen if not.

**Comment by grantila** [comment]
Using 2.24.0 I still get this at `import type { Node as ASTNode } from 'unist';` when `@types/unist` is installed, but not `unist` itself. Using yarn 1 with workspaces.

**Comment by ljharb** [comment]
@grantila can you provide a repro repo?
